sdk/tests/core/CMakeLists.txt

```diff
@@ -24,13 +31,15 @@ endif()
 if (AZ_PLATFORM_IMPL STREQUAL "POSIX" AND NOT APPLE)
     add_cmocka_test(az_core_test SOURCES
                     main.c
+                    mock_az_mqtt.c
```

Comment:
does `_az_MOCK_ENABLED` need to be checked here as well?

Reply:
Thanks for pointing it out, only `UNIT_TESTING_MOCKS` should be checked, `az_MOCK_ENABLED` is a compile time definition